### PR TITLE
Components/tag

### LINKED ISSUE #7: Tag
Bildschirmfoto 2021-11-15 um 11 49 50

### PR DESCRIPTION
**Pull request by iva585**
Tag Component

closes #7 

We've built the tag component, which allows the user to filter movie results. 
It is able to show the count of movies in each category, but also not show any count (see tag ALL).

Different tag versions can be seen in Storybook.

Tag

Tagtag2

### CODE REVIEW

File: src/app/components/Tag/Tag.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+export default function Tag({
+  children,
+  tagStatus,
+  numberOfItems,
+}: TagProps): JSX.Element {
+  return (
+    <TagStyleActive tagStatus={tagStatus}>{`${children} ${
+      numberOfItems ? numberOfItems : ''
+    }`}</TagStyleActive>
```
Comment (phlgr) [inline src/app/components/Tag/Tag.tsx:19]:
```suggestion
    <TagStyleActive active={active}>{children} {count && ({count})}</TagStyleActive>
```

File: src/app/components/Tag/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  );
+}
+const TagStyleActive = styled.span<Partial<TagProps>>`
+  background-color: ${(props) => (props.tagStatus ? '#FFC700' : '#2A2A2A')};
```
Comment (phlgr) [inline src/app/components/Tag/Tag.tsx:23]:
```suggestion
  background-color: ${(props) => (props.active? '#FFC700' : '#2A2A2A')};
```

File: src/app/components/Tag/Tag.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import type { ReactNode } from 'react';
+import styled from 'styled-components';
+
+type TagProps = {
+  children: ReactNode;
+  tagStatus?: boolean;
+  numberOfItems?: number;
```
Comment (phlgr) [inline src/app/components/Tag/Tag.tsx:8]:
```suggestion
  active?: boolean;
  count?: string;
```

File: src/app/components/Tag/Tag.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export default function Tag({
+  children,
+  tagStatus,
+  numberOfItems,
```
Comment (phlgr) [inline src/app/components/Tag/Tag.tsx:14]:
```suggestion
  active = false,
  count,
```

File: src/app/components/Tag/Tag.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+export default {
+  title: 'Component/Tag',
+  component: Tag,
+};
+
+export const NotActive = (): JSX.Element => (
+  <Tag numberOfItems={3} tagStatus={false}>
+    Action
+  </Tag>
```
Comment (phlgr) [inline src/app/components/Tag/Tag.stories.tsx:12]:
```suggestion
  <Tag count="3">
    Action
  </Tag>
```

File: src/app/components/Tag/Tag.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+);
+export const Active = (): JSX.Element => (
+  <Tag numberOfItems={3} tagStatus={true}>
+    Action
+  </Tag>
```
Comment (phlgr) [inline src/app/components/Tag/Tag.stories.tsx:17]:
```suggestion
  <Tag count="3" active>
    Action
  </Tag>
```

File: src/app/components/Tag/Tag.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+);
+export const All = (): JSX.Element => <Tag tagStatus={true}>Action</Tag>;
```
Comment (phlgr) [inline src/app/components/Tag/Tag.stories.tsx:19]:
```suggestion
export const All = (): JSX.Element => <Tag active>Action</Tag>;
```

**Review by phlgr — CHANGES_REQUESTED** [review]
Nice work! 👾

**Review by marwinburesch — APPROVED** [review]
👍 noice

**Pull request merged**